src/main/java/org/skriptlang/skript/elements/fireworks/ExprFireworkFlownTime.java

```diff
@@ -0,0 +1,159 @@
+		"\tadd 2 seconds to time until detonation of event-firework"
+})
+@Since("INSERT VERSION")
+@SuppressWarnings("removal")
+public class ExprFireworkFlownTime extends SimplePropertyExpression<Firework, Timespan> {
+
+	private final static boolean PAPER = Skript.methodExists(Firework.class, "getTicksFlown");
+
+	/**
+	 * Developer note:
+	 * Spigot's methods are {@link Firework#getLife()} and {@link Firework#getMaxLife()} and were added in 1.19.
+	 * Paper has had these methods since 1.18 but were named the following;
+	 * {@link Firework#getTicksFlown()} and {@link Firework#getTicksToDetonate()} respectfully.
+	 * Paper does not agree with Spigot's method naming, so they have Spigot's deprecated.
+	 */
+	static {
+		registerDefault(ExprFireworkFlownTime.class, Timespan.class, "firework ([detonate:max[imum]] life[time]|flown time|detonate:time until detonat(e|ion))", "fireworks");
+	}
+
+	private boolean detonate;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		detonate = parseResult.hasTag("detonate");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public Timespan convert(Firework firework) {
+		if (PAPER)
+			return Timespan.fromTicks_i(detonate ? firework.getTicksToDetonate() : firework.getTicksFlown());
+		return Timespan.fromTicks_i(detonate ? firework.getMaxLife() : firework.getLife());
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		return (mode != ChangeMode.REMOVE_ALL) ? CollectionUtils.array(Timespan.class) : null;
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		long ticks = ((Timespan) delta[0]).getTicks_i();
+		switch (mode) {
+			case REMOVE:
+				ticks = -ticks;
+			case ADD:
+				for (Firework firework : getExpr().getArray(event)) {
+					long existing = convert(firework).getTicks_i();
+					if (detonate) {
+						if (PAPER) {
+							firework.setTicksToDetonate((int) Math.min(0, existing + ticks));
+						} else {
+							firework.setMaxLife((int) Math.min(0, existing + ticks));
+						}
+					} else {
+						if (PAPER) {
+							firework.setTicksFlown((int) Math.min(0, existing + ticks));
+						} else {
+							firework.setLife((int) Math.min(0, existing + ticks));
+						}
+					}
+				}
+				break;
+			case SET:
+				for (Firework firework : getExpr().getArray(event)) {
+					if (detonate) {
+						if (PAPER) {
+							firework.setTicksToDetonate((int) Math.min(0, ticks));
+						} else {
+							firework.setMaxLife((int) Math.min(0, ticks));
+						}
+					} else {
+						if (PAPER) {
+							firework.setTicksFlown((int) Math.min(0, ticks));
+						} else {
+							firework.setLife((int) Math.min(0, ticks));
+						}
+					}
+				}
+				break;
+			case DELETE:
+			case REMOVE_ALL:
+			case RESET:
+				for (Firework firework : getExpr().getArray(event)) {
```

Comment:
If `REMOVE_ALL` isn't a valid changer we shouldn't include it under delete and reset but rather `default` with `asset false`